### ISSUE TITLE
Visual quirk with button groups with a border radius

### ISSUE THREAD

**Issue by gakimball** [zurb/foundation-sites#7665]
Reopening issue 172 by @ProLoser from the old foundation-sites-6 repo.

> So the fact that in a button group all the right borders are white (in order to create the gap) creates a weird visual quirk with the border-radius. I realize that inline-block is not normally super susceptible to margins, but in this case I think using a 1px margin instead of a white border would be smarter.

> I found a lot of frustrations with how borders were utilized in f5 and was required to completely rewrite the buttons mixins because borders are explicitly used to create buttons, even though they don't appear visually on the screen. The problem with this is that if you actually WANT borders to appear on buttons, you have to do a LOT of css overriding and it was just easier to write a lighter mixin.

> Any thoughts on removing border css from buttons? Should I file another ticket?

@ProLoser's suggestion for stripping radius properties from inner buttons:

```css
.button-group .button:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.button-group .button:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
```

![image](https://cloud.githubusercontent.com/assets/1142001/11968365/717153c4-a8c0-11e5-940c-fb261622b461.png)

![image](https://cloud.githubusercontent.com/assets/1142001/11968380/9ae778a0-a8c0-11e5-97f8-74b9aa8f196a.png)

**Comment by marcusconnor** [comment]
I agree with using margin rather than border as a separator. I also think the stacked group should follow the same convention with a single pixel margin on the bottom of each button. The current border separator doesn't look good in default because it causes weird visual behavior.

This image shows current default behavior:

![button-group-issue](https://cloud.githubusercontent.com/assets/11634985/12208456/d059c3dc-b61a-11e5-9985-dcccf9544684.png)

**Comment by gakimball** [comment]
Our button groups are made with `display: table` and `display: table-cell`, and it appears that setting `margin-right` doesn't do anything—the margin doesn't apply, and the buttons sit right next to each other.

**Comment by andycochran** [comment]
This is another reason to maybe rethink the `display:table` approach for button groups—in addition to #7844 (`<button>` element not working).

**Comment by gakimball** [comment]
@andycochran Even with the old approach where we used floats and percentage widths, it still wouldn't work, because margin doesn't factor into width calculation. Is there another way to do even width?

**Comment by andycochran** [comment]
@gakimball Ooh! What about using [calc()](http://caniuse.com/#search=calc)? Something like… 

```css
.button-group.expanded-4 .button {
    display: inline-block;
    width: calc(25% - 1px);
    margin-right: 1px;
}
```

**Comment by gakimball** [comment]
@andycochran That would work, but we'd need to use floats, because `display: inline-block` adds extra whitespace.

**Comment by andycochran** [comment]
@gakimball If we also do `.button-group { font-size: 0; }`, there'd be no whitespace.

**Comment by gakimball** [comment]
@andycochran Actually, wait, `calc()` isn't supported in Android 2 :/

**Comment by andycochran** [comment]
@gakimball Ack! I'd just checked 4. 
FWIW, only 3.2% of Androids are running 2.

**Comment by gakimball** [comment]
@andycochran Android 2.3 specifically is our cutoff because it has marginally more use than older versions of Android 2.

It's a hassle for sure, but we didn't plan on adjusting our browser compatibility between v5 and v6. Maybe for v7... whenever that is... we can take a look again. Android 2 *mostly* hasn't been a hassle, but there have been small issues.

**Comment by andycochran** [comment]
@gakimball I don't mean to suggest dropping browser support. But buttons looking nice in every other browser—and `<button>` elements working (see the above reference)—seems like a good reason to drop the `display:table`, even w/ Android 2 taking a hit.

**Comment by gakimball** [comment]
@andycochran Alright, then if we can get the desired effect on new browsers, and have it look a little janky (but not broken) on Android 2, that should be fine.

**Comment by andycochran** [comment]
Cool. I'll take a stab at a PR…

**Comment by gakimball** [comment]
Fixed via #7933
